tests/users/test_data.py

```diff
@@ -174,3 +174,19 @@ async def test_find_or_create_b2c_user(
         assert int(user.handle.split("-")[-1])
 
     assert user == snapshot(matcher=path_type({"handle": (str,)}))
+
+
+async def test_check_user_does_not_exist(users_data):
```

Comment:
I think parametrization is an ideal solution here rather than having separate tests.

```suggestion
async def test_check_no_users_exist(users_data):
```